src/Gui/TextEdit.cpp

```diff
@@ -448,8 +448,15 @@ void TextEditor::OnChange(Base::Subject<const char*> &rCaller,const char* sReaso
 #else
         int fontSize = hPrefGrp->GetInt("FontSize", 10);
 #endif
-        QFont font = QFontDatabase::systemFont(QFontDatabase::FixedFont);
-        font.setPointSize(fontSize);
+        QFont font;
+        auto fontName = hPrefGrp->GetASCII("Font", "None");

```

Comment:
Same suggestion as above: Omitt the second argument and then check `if (fontName.empty())`.